tests/testthat/test-backend-dbplyr__duckdb_connection.R

```diff
@@ -129,6 +129,22 @@ test_that("custom lubridate functions translated correctly", {
   expect_equal(translate(floor_date(x, "week", week_start = 4)), sql(r"{CAST(x AS DATE) - CAST(EXTRACT('dow' FROM CAST(x AS DATE) + 3) AS INTEGER)}"))
 })
 
+# clock functions
+test_that("custom clock functions translated correctly", {
+  skip_if_no_R4()
+  skip_if_not_installed("dbplyr")
+  con <- dbConnect(duckdb())
+  on.exit(dbDisconnect(con, shutdown = TRUE))
+  translate <- function(...) dbplyr::translate_sql(..., con = con)
+  sql <- function(...) dbplyr::sql(...)
+
+  expect_equal(translate(add_days(x, 1L)), sql(r"{date_add(x, INTERVAL '1 day')}"))
+  expect_equal(translate(add_years(x, 2L)), sql(r"{date_add(x, INTERVAL '2 day')}"))
```

Comment:
```suggestion
  expect_equal(translate(add_days(x, 2L)), sql(r"{date_add(x, INTERVAL '2 day')}"))
```

What happens with negative days? It's also interesting to see an integration test -- running an actual query with negative months, let's say.

Reply:
@krlmlr sure, here's a reprex (comparing clock's behaviour for a dataframe with how this duckdb implementation works) and it seems to work as expected for both adding days and years as well as substracting them

``` r
options(conflicts.policy = list(warn = FALSE))
library(DBI)
#> Warning: package 'DBI' was built under R version 4.2.3
library(RPostgres)
#> Warning: package 'RPostgres' was built under R version 4.2.3
library(dplyr)
#> Warning: package 'dplyr' was built under R version 4.2.3
library(dbplyr)
#> Warning: package 'dbplyr' was built under R version 4.2.3
library(clock)
#> Warning: package 'clock' was built under R version 4.2.3
library(duckdb)

con <- DBI::dbConnect(duckdb(), path = ":memory:")
test_data <- data.frame(
  person = 1L,
  date_1 = as.Date("2000-01-01")
)
test_data |> 
  mutate(date_plus_two_days  = clock::add_days(date_1, 2), 
         date_plus_two_years = clock::add_years(date_1, 2),
         date_minus_two_days  = clock::add_days(date_1, -2), 
         date_minus_two_years = clock::add_years(date_1, -2)) |> 
  dplyr::glimpse()
#> Rows: 1
#> Columns: 6
#> $ person               <int> 1
#> $ date_1               <date> 2000-01-01
#> $ date_plus_two_days   <date> 2000-01-03
#> $ date_plus_two_years  <date> 2002-01-01
#> $ date_minus_two_days  <date> 1999-12-30
#> $ date_minus_two_years <date> 1998-01-01

db_test_data <- copy_to(con, test_data, overwrite = TRUE)
db_test_data |> 
  mutate(date_plus_two_days  = clock::add_days(date_1, 2), 
         date_plus_two_years = clock::add_years(date_1, 2),
         date_minus_two_days  = clock::add_days(date_1, -2), 
         date_minus_two_years = clock::add_years(date_1, -2)) |> 
  dplyr::glimpse()
#> Rows: ??
#> Columns: 6
#> Database: DuckDB v0.10.3-dev163 [eburn@Windows 10 x64:R 4.2.1/:memory:]
#> $ person               <int> 1
#> $ date_1               <date> 2000-01-01
#> $ date_plus_two_days   <dttm> 2000-01-03
#> $ date_plus_two_years  <dttm> 2002-01-01
#> $ date_minus_two_days  <dttm> 1999-12-30
#> $ date_minus_two_years <dttm> 1998-01-01
```

<sup>Created on 2024-05-03 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
Thanks for the example. Having a code like this in a test would be useful too.